VMEncryption/main/CryptMountConfigUtil.py

```diff
@@ -610,3 +621,40 @@ def restore_mount_info(self, mount_point_or_mapper_name):
             f.write('\n')
 
         self.logger.log("fstab updated successfully")
+
+    def migrate_crypt_items(self):
+        with open('/etc/fstab', 'r') as f:
+            lines = f.readlines()
+        
+        if not self.is_bek_in_fstab_file(lines):
+            self.logger.log("BEK volume not detected in fstab. Adding it now.")
+            lines.append(self.get_fstab_bek_line())
+            self.add_bek_to_default_cryptdisks()
+            with open('/etc/fstab', 'w') as f:
+                f.writelines(lines)
+
+        crypt_items = self.get_crypt_items()
+
+        for crypt_item in crypt_items:
+            self.logger.log("Migrating crypt item: {0}".format(crypt_item))
+            if crypt_item.mount_point == "/" or CommonVariables.osmapper_name == crypt_item.mapper_name:
+                self.logger.log("Skipping OS disk")
+                continue
+
+            if crypt_item.mapper_name and crypt_item.mount_point and crypt_item.mount_point != "None":
+                self.logger.log(msg="Adding entry for {0} drive in fstab with mount point {1}".format(crypt_item.mapper_name, crypt_item.mount_point), level=CommonVariables.InfoLevel)
+                self.append_mount_info_data_disk(os.path.join(CommonVariables.dev_mapper_root, crypt_item.mapper_name), crypt_item.mount_point)
```

Comment:
Is it feasible to quickly check here that the crypt_item.mapper_name does indeed refer to a mountable file system?

Reply:
@vermashi Added logic to check if mapper path exists before adding them to crypttab/fstab

Reply:
I think the actual ask here was to check if the mapper device has a mountable file system.
So in addition to checking that device_path exists, add a way to test that it has a file system:
you can do one of the following:
 (1) Try to mount the device and check if it got mounted
 (2) Use disk_util.get_device_items on the device path and see if its file systems are one of the known supported file systems (xfs, ext2,3,4)

Reply:
Done